packages/auth/src/client/service.test.ts

```diff
@@ -43,6 +61,400 @@ describe('Client Service', (): void => {
     }
   )
 
+  describe('signatures', (): void => {
+    test('can verify a signature', async (): Promise<void> => {
+      const challenge = 'test-challenge'
+      const privateJwk = (await importJWK(TEST_PRIVATE_KEY)) as crypto.KeyLike
+      const signature = crypto.sign(null, Buffer.from(challenge), privateJwk)
+      const verified = await clientService.verifySig(
+        signature.toString('base64'),
+        TEST_PUBLIC_KEY,
+        challenge
+      )
+
+      expect(verified).toBe(true)
+    })
+
+    test('can construct a challenge from signature input', (): void => {
+      const sigInputHeader =
+        'sig1=("@method" "@target-uri" "content-digest" "content-length" "content-type" "authorization");created=1618884473;keyid="gnap-key"'
+      const ctx = createContext(
+        {
+          headers: {
+            'Content-Type': 'application/json',
+            'Content-Digest': 'sha-256=:test-hash:',
+            'Content-Length': '1234',
+            'Signature-Input': sigInputHeader,
+            Authorization: 'GNAP test-access-token'
+          }
+        },
+        {}
+      )
+
+      ctx.request.body = { foo: 'bar' }
+      ctx.method = 'GET'
+      ctx.request.url = '/test'
```

Comment:
Maybe add a function that takes `sigInputHeader` and optional `authorization` header that handles all this setup for all these tests.
Or combine all these into a single `test.each`